contracts/utils/Votes.sol

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./Context.sol";
+import "./Counters.sol";
+import "./Checkpoints.sol";
+import "../interfaces/IVotes.sol";
+import "./cryptography/draft-EIP712.sol";
+
+/**
+ * @dev Voting operations.
+ *
+ * This extension keeps a history (checkpoints) of each account's vote power. Vote power can be delegated either
+ * by calling the {delegate} function directly, or by providing a signature to be used with {delegateBySig}. Voting
+ * power can be queried through {getVotes}.
+ *
+ * By default, token balance does not account for voting power. This makes transfers cheaper. The downside is that it
+ * requires users to delegate to themselves in order to activate checkpoints and have their voting power tracked.
+ * Enabling self-delegation can easily be done by overriding the {delegates} function. Keep in mind however that this
+ * will significantly increase the base gas cost of transfers.
+ *
+ * When using this module, the derived contract must implement {_getDelegatorVotingPower}, and can use {_moveVotingPower}
+ * when a delegator's voting power is changed.
+ */
+abstract contract Votes is Context, EIP712, IVotes {
+    using Checkpoints for Checkpoints.History;
+    using Counters for Counters.Counter;
+
+    bytes32 private constant _DELEGATION_TYPEHASH =
+        keccak256("Delegation(address delegatee,uint256 nonce,uint256 expiry)");
+    mapping(address => address) private _delegateCheckpoints;
+    mapping(address => Checkpoints.History) private _userCheckpoints;
+    mapping(address => Counters.Counter) private _nonces;
```

Comment:
I'd prefer seeing that line after the _totalCheckpoints so that the 3 slots used for delegation:
```
mapping(address => address) private _delegate;
mapping(address => Checkpoints.History) private _userCheckpoints;
Checkpoints.History private _totalCheckpoints;
```

are consecutive.